cmd/corald/midware/auth.go

```diff
@@ -0,0 +1,118 @@
+package midware
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/ardanlabs/kit/cfg"
+	"github.com/ardanlabs/kit/log"
+	"github.com/ardanlabs/kit/web/app"
+	"github.com/dgrijalva/jwt-go"
+)
+
+var (
+	// ErrInvalidToken is returned when the token provided is not valid.
+	ErrInvalidToken = errors.New("invalid token")
+
+	// ErrInvalidClaims is returned when the claims inside a valid token are not
+	// valid.
+	ErrInvalidClaims = errors.New("invalid claims")
+)
+
+const cfgAuthPublicKey = "AUTH_PUBLIC_KEY"
+
+// authOff is used when authentication is turned off by not providing a public
+// key in the environment.
+func authOff(h app.Handler) app.Handler {
+	return func(c *app.Context) error {
+
+		// Log out the process for verbosity.
+		log.Dev(c.SessionID, "Auth", "Started")
+		log.Dev(c.SessionID, "Auth", "Authentication Off")
+		log.Dev(c.SessionID, "Auth", "Completed")
+		return h(c)
+	}
+}
+
+// Auth handles token authentication.
+func Auth() (app.Middleware, error) {
+
+	// Load in the public key to validate the JWT tokens.
+	publicKeyBase64Str, err := cfg.String(cfgAuthPublicKey)
+	if err != nil {
+		return authOff, nil
+	}
+
+	publicKeyPEM, err := base64.StdEncoding.DecodeString(publicKeyBase64Str)
```

Comment:
Can you describe what you are doing here and why? The workflow is what is important.